.github/files/gh-wp-svn-autopublish/files/wp-svn-autopublish.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+
+set -eo pipefail
+
+: "${GITHUB_REF:?Build argument needs to be set and non-empty.}"
+: "${WPSVN_USERNAME:?Build argument needs to be set and non-empty.}"
+: "${WPSVN_PASSWORD:?Build argument needs to be set and non-empty.}"
+
+## Determine tag
+if [[ ! "$GITHUB_REF" =~ ^refs/tags/v?[0-9]+(\.[0-9]+)+(-[a-z0-9._-]+)?$ ]]; then
+	echo "::error::Expected GITHUB_REF like \`refs/tags/v1.2.3\` or \`refs/tags/1.2.3\`, got \`$GITHUB_REF\`"
+	exit 1
+fi
+TAG="${GITHUB_REF#refs/tags/}"
+TAG="${TAG#v}"
+
+## Determine slug
+WPSLUG=$(jq -r '.extra["wp-plugin-slug"] // ""' "src/composer.json")
+if [[ -z "$WPSLUG" ]]; then
+	echo '::error::Failed to determine plugin slug.'
+	exit 1
+fi
+
+echo "Publishing $WPSLUG version $TAG"
+
+mkdir svn
+cd svn
+
+echo '::group::Checking out SVN (shallowly)'
+svn checkout "https://plugins.svn.wordpress.org/$WPSLUG/" --depth=empty .
+echo '::endgroup::'
+
+echo '::group::Checking out SVN trunk'
+svn up trunk
+echo '::endgroup::'
+
+echo "::group::Checking out SVN tags (shallowly)"
+svn up tags --depth=empty
+echo '::endgroup::'
+
+echo "::group::Deleting everything in trunk except for .svn directories"
+find trunk ! \( -path '*/.svn/*' -o -path "*/.svn" \) \( ! -type d -o -empty \) -delete
+[[ -e trunk ]] || mkdir -p trunk # If there were no .svn directories, trunk itself might have been removed.
+echo '::endgroup::'
+
+echo "::group::Copying git repo into trunk"
+git clone ../src trunk/
```

Comment:
I'm trying to think through issues with this.

Say, security release and putting out a lot of different versions, this could cause issues with trunk changing for each release as we're tagging it, making the stable change (unless we ensure that every branch has the right stable value on it, which I don't know if we can trust).

@jeherve what do you think?

Should we have some type of check if the stable version line would change in trunk or just ensure that line doesn't change? I'm mulling it over.

Reply:
Yeah, let's do that. 522a7a2a95